lib/remote-connector.js

```diff
@@ -104,16 +104,29 @@ function createProxyMethod(Model, remotes, remoteMethod) {
       callback = utils.createPromiseCallback();
     }
 
-    var reqOptions
-    var lastArg = args[args.length - 1]
-    if (
-      typeof lastArg === 'object' &&
-      lastArg &&
-      'reqOptions' in lastArg &&
-      typeof lastArg.reqOptions === 'object' &&
-      lastArg.reqOptions
-    ) {
-      reqOptions = args.pop().reqOptions
+    // Support taking reqOptions off the callback, or off the last arg (after the callback)
+    // which is commonly an options object - this allows it to be passed without changing the
+    // method call signature, which might have unintended consequences when a remote method is
+    // invoked in a non-remote context
+    var reqOptions;
+    if (lastArgIsFunc) {
+      if (typeof callback.reqOptions && callback.reqOptions) {
+        reqOptions = callback.reqOptions;
+        delete callback.reqOptions;
+      }
+    }
+
+    if (!reqOptions) {
+      var lastArg = args[args.length - 1];
+      if (
+        typeof lastArg === 'object' &&
+        lastArg &&
+        typeof lastArg.reqOptions === 'object' &&
+        lastArg.reqOptions
+      ) {
+        reqOptions = lastArg.reqOptions
+        delete lastArg.reqOptions;
+      }
     }
```

Comment:
Given this is pulling `reqOptions` off the last arg, regardless of what the last Arg is, should we make it use a symbol instead of a known object key?